src/Randomizer.SMZ3.Tracking/AutoTracking/AutoTracker.cs

```diff
@@ -0,0 +1,864 @@
+            {
+                var region = Tracker.World.Regions.First(x => dungeonInfo.Is(x) && x is Z3Region) as Z3Region;
+                if (region == null)
+                {
+                    _logger.LogError($"Could not find region for {dungeonInfo.Name}");
+                    Tracker.Error();
+                    continue;
+                }
+
+                try
+                {
+                    if (!dungeonInfo.Cleared && data.CheckUInt16(region.MemoryAddress * 2 ?? 0, region.MemoryFlag ?? 0))
+                    {
+                        Tracker.MarkDungeonAsCleared(dungeonInfo);
+                        _logger.LogInformation($"Auto tracked {dungeonInfo.Name} as cleared");
+                    }
+
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(e, "Unable to auto track Dungeon: " + dungeonInfo.Name);
+                    Tracker.Error();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Checks the status of if the Super Metroid bosses have been defeated
+        /// </summary>
+        /// <param name="message">The response from the lua script</param>
+        protected void CheckSMBosses(EmulatorMemoryData data)
+        {
+            var boss = Tracker.WorldInfo.Bosses.First(x => "Kraid".Equals(x.Name[0], StringComparison.OrdinalIgnoreCase));
+            if (!boss.Defeated && data.CheckBinary8Bit(0x1, 0x1))
+            {
+                Tracker.MarkBossAsDefeated(boss);
+                _logger.LogInformation($"Auto tracked {boss.Name} as defeated");
+            }
+
+            boss = Tracker.WorldInfo.Bosses.First(x => "Ridley".Equals(x.Name[0], StringComparison.OrdinalIgnoreCase));
+            if (!boss.Defeated && data.CheckBinary8Bit(0x2, 0x1))
+            {
+                Tracker.MarkBossAsDefeated(boss);
+                _logger.LogInformation($"Auto tracked {boss.Name} as defeated");
+            }
+
+            boss = Tracker.WorldInfo.Bosses.First(x => "Phantoon".Equals(x.Name[0], StringComparison.OrdinalIgnoreCase));
+            if (!boss.Defeated && data.CheckBinary8Bit(0x3, 0x1))
+            {
+                Tracker.MarkBossAsDefeated(boss);
+                _logger.LogInformation($"Auto tracked {boss.Name} as defeated");
+            }
+
+            boss = Tracker.WorldInfo.Bosses.First(x => "Draygon".Equals(x.Name[0], StringComparison.OrdinalIgnoreCase));
+            if (!boss.Defeated && data.CheckBinary8Bit(0x4, 0x1))
+            {
+                Tracker.MarkBossAsDefeated(boss);
+                _logger.LogInformation($"Auto tracked {boss.Name} as defeated");
+            }
+        }
+
+        /// <summary>
+        /// Tracks the current memory state of LttP for Tracker voice lines
+        /// </summary>
+        /// <param name="message">The message from the emulator with the memory state</param>
+        protected void CheckZeldaState(EmulatorAction action)
```

Comment:
Yeah, I've had the same thoughts. I could possibly implement something like CS-Script and potentially a new config for all these where you basically put the if condition in there and what tracker can say as a result. This could allow people to add in their own potentially (especially if I add a debugger window that prints out the current Zelda/Metroid states).

Reply:
I'm not too sure about scripting because that could very easily lead to abuse.

I want to come up with something slightly complicated to add custom conditions to Tracker lines (e.g. only if go mode is enabled, or only if chat is connected). Whatever system that ends up being (probably something naive), I could probably apply to this, too.

Reply:
That's fair. I feel like that could get pretty unwieldy in and of itself with all these Zelda/Metroid state properties, but maybe I'm just not picturing it all in my head super well right now.